src/System.IO.Abstractions.TestingHelpers/MockDirectoryInfo.cs

```diff
@@ -114,29 +114,29 @@ public override string FullName
         /// <inheritdoc />
         public override DateTime LastAccessTime
         {
-            get { return GetMockFileDataForRead().LastAccessTime.DateTime; }
+            get { return GetMockFileDataForRead().LastAccessTime.LocalDateTime; }
```

Comment:
These changes make #879 obsolete, right?

Reply:
Yeah, more or less.
It changes the behaviour, that `DateTimeKind.Unspecified` is now consistently interpreted as local time...

Reply:
@xtopaz Can you double check that this changes also solves the issue you have been seeing?

Reply:
No, the changes unfortunately do not fix the bug described in #879. 
Since in the timestamp setters the value is always implicitly converted to `DateTimeOffset` and thus `DateTimeKind.Unspecified` is treated as local, there is a further time shift here.

Reply:
OK, now it gets complicated @xtopaz .
Your proposed solution works, when you read/write UTC times (e.g. `LastAccessTimeUtc`), that they remain unchanged when providing `DateTimeKind.Unspecified`.
My solution works, when you read/write the local times (e.g. `LastAccessTime`), that they remain unchanged when providing `DateTimeKind.Unspecified`.
However the real file system works in both cases, that it doesn't change the time, but only adjusts the `DateTimeKind`, so both the following tests against the real file system are successful:

```csharp
        [Test]
        public void RealFileInfo_LastAccessTimeUtcWithUnspecifiedDateTimeKind_ShouldSetLastAccessTimeUtcOfFileInFileSystem()
        {
            var date = DateTime.SpecifyKind(DateTime.Now.AddHours(-4), DateTimeKind.Unspecified);
            var fileSystem = new FileSystem();
            fileSystem.Directory.CreateDirectory(@"c:\test");
            fileSystem.File.WriteAllText(@"c:\test\a.txt", "Demo text content");
            var fileInfo = new FileInfo(@"c:\test\a.txt")
            {
                LastAccessTimeUtc = date
            };

            Assert.AreEqual(date, fileInfo.LastAccessTimeUtc);
            Assert.AreNotEqual(DateTimeKind.Unspecified, fileInfo.LastAccessTimeUtc);
        }
        [Test]
        public void RealFileInfo_LastAccessTimeWithUnspecifiedDateTimeKind_ShouldSetLastAccessTimeOfFileInFileSystem()
        {
            var date = DateTime.SpecifyKind(DateTime.Now.AddHours(-4), DateTimeKind.Unspecified);
            var fileSystem = new FileSystem();
            fileSystem.Directory.CreateDirectory(@"c:\test");
            fileSystem.File.WriteAllText(@"c:\test\a.txt", "Demo text content");
            var fileInfo = new FileInfo(@"c:\test\a.txt")
            {
                LastAccessTime = date
            };

            Assert.AreEqual(date, fileInfo.LastAccessTime);
            Assert.AreNotEqual(DateTimeKind.Unspecified, fileInfo.LastAccessTime);
        }
```

Reply:
@fgreinacher : I would suggest to close this pull request and fix the behaviour with `DateTimeKind.Unspecified` explicitely with #879 .
I could provide or help with a pull request (based on my changes here), that implements the same behaviour as the real file system in both cases...

Reply:
The fix implemented in #879 already fixes the behavior in `MockFileInfo`, but `FileInfoWrapper` is not affected by the bug described there.

Please check https://github.com/TestableIO/System.IO.Abstractions/pull/879#discussion_r969527170

Reply:
> I would suggest to close this pull request

What about just removing the changes related to timezone handling from this PR? The changes that update the timestamps look good to me?

Reply:
@fgreinacher:
It would have been harder to remove the timezone handling from this PR, than to also fix #879 here.
I included the tests from @xtopaz from #879 and they ran successful.

Reply:
Great, thanks to both of you @xtopaz and @vbreuss 👍

I'll merge this one. I think we don't need #879 anymore then, right?

Reply:
@fgreinacher Yes, i've closed #879.